proxy_agent/src/common/http/request.rs

```diff
@@ -200,24 +200,43 @@ impl Request {
 
         let path = String::from(url.path());
 
-        let parameters = url.query_pairs();
-        let mut pairs: HashMap<String, String> = HashMap::new();
-        let mut canonicalized_parameters = String::new();
-        if parameters.count() > 0 {
-            for p in parameters {
-                // Convert the parameter name to lowercase
-                pairs.insert(p.0.to_lowercase(), p.1.to_string());
+        // The query_pairs method from the Url struct returns an iterator over the query pairs, which are automatically percent-decoded
+        // To get the raw percent-encoded values, we need to manually parse the query string
+        let query = url.query().unwrap_or("");
```

Comment:
instead of using unwrap, can you use match and log if error

Reply:
query() returns Option<&str>, we already logged the Uri when received the request.